engines/dragons/actor.cpp

```diff
@@ -0,0 +1,794 @@
+		debug("Unable to find free actor slot!!");
+	}
+	resetDisplayOrder();
+	return actor;
+}
+
+Actor *ActorManager::findFreeActor(int16 resourceId) {
+	int i = 0;
+	for (ActorsIterator it = _actors.begin(); it != _actors.end() && i < 23; ++it, i++) {
+		Actor *actor = it;
+		if (!(actor->_flags & Dragons::ACTOR_FLAG_40)) {
+			actor->resourceID = resourceId;
+			actor->_walkSpeed = 0x100000;
+			return actor;
+		}
+	}
+	return NULL;
+}
+
+Actor *ActorManager::getActor(uint16 actorId) {
+	assert(actorId < DRAGONS_ENGINE_NUM_ACTORS);
+	return &_actors[actorId];
+}
+
+void ActorManager::clearActorFlags(uint16 startingActorId) {
+	assert(startingActorId < DRAGONS_ENGINE_NUM_ACTORS);
+	for(uint16 i = startingActorId; i < DRAGONS_ENGINE_NUM_ACTORS; i++) {
+		_actors[i]._flags = 0;
+	}
+}
+
+Actor *ActorManager::loadActor(uint32 resourceId, uint16 actorId) { //TODO should we rename this. loadActorResource or updateActorResource
+	Actor *actor = getActor(actorId);
+	actor->_actorResource = _actorResourceLoader->load(resourceId);
+	return actor;
+}
+
+ActorResource *ActorManager::getActorResource(uint32 resourceId) {
+	return _actorResourceLoader->load(resourceId);
+}
+
+void ActorManager::updateActorDisplayOrder() {
+	bool shouldContinue = true;
+
+	while(shouldContinue) {
+		shouldContinue = false;
+		for (int i = 0; i < DRAGONS_ENGINE_NUM_ACTORS - 1; i++) {
+			Actor *curActor = getActor(_displayOrder[i]);
+			Actor *nextActor = getActor(_displayOrder[i + 1]);
+			int16 curY = curActor->y_pos > 0 ? curActor->y_pos : 0;
+			int16 nextY = nextActor->y_pos > 0 ? nextActor->y_pos : 0;
+			if (nextActor->priorityLayer * 0x1000000 + nextY * 0x100 + nextActor->_actorID <
+				curActor->priorityLayer * 0x1000000 + curY * 0x100 + curActor->_actorID) {
+				_displayOrder[i] = nextActor->_actorID;
+				_displayOrder[i + 1] = curActor->_actorID;
+				shouldContinue = true;
+			}
+		}
+	}
+}
+
+void ActorManager::resetDisplayOrder() {
+	for (uint16 i = 0; i < DRAGONS_ENGINE_NUM_ACTORS; i++) {
+		Actor *actor = getActor(i);
+		_displayOrder[i] = i;
+		if (!actor->isFlagSet(ACTOR_FLAG_40)) {
+			actor->priorityLayer = 0;
+		}
+	}
+}
+
+Actor *ActorManager::getActorByDisplayOrder(uint16 position) {
+	return getActor(_displayOrder[position]);
+}
+
+Actor::Actor(uint16 id) : _actorID(id) {
+	_actorResource = NULL;
+	resourceID = -1;
+	_seqCodeIp = 0;
+	frame_pointer_maybe = NULL;
+	priorityLayer = 3;
+	x_pos = 160;
+	y_pos = 110;
+	_walkDestX = 0;
+	_walkDestY = 0;
+	_walkSpeed = 0;
+	_flags = 0;
+	frame_width = 0;
+	frame_height = 0;
+	frame_flags = 0;
+	clut = 0;
+	frame = NULL;
+	surface = NULL;
+}
+
+void Actor::init(ActorResource *resource, int16 x, int16 y, uint32 sequenceID) {
+	debug(3, "actor %d Init", _actorID);
+	_actorResource = resource;
+	x_pos = x;
+	y_pos = y;
+	sequenceTimer = 0;
+	_walkDestX = x;
+	_walkDestY = y;
+	scale = DRAGONS_ENGINE_SPRITE_100_PERCENT_SCALE;
+	_sequenceID2 = 0;
+	_flags = (Dragons::ACTOR_FLAG_40 | Dragons::ACTOR_FLAG_4);
```

Comment:
You are already within `Dragons` namespace. Is the verbatim namespace necessary here?